src/main/java/org/radarbase/management/web/rest/KratosEndpoint.kt

```diff
@@ -0,0 +1,109 @@
+import org.radarbase.auth.kratos.SessionService
+import org.radarbase.management.config.ManagementPortalProperties
+import org.radarbase.management.repository.SubjectRepository
+import org.radarbase.management.security.NotAuthorizedException
+import org.radarbase.management.service.*
+import org.radarbase.management.service.dto.KratosSubjectWebhookDTO
+import org.radarbase.management.web.rest.errors.EntityName
+import org.radarbase.management.web.rest.errors.NotFoundException
+import org.radarbase.management.web.rest.util.HeaderUtil
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+
+@RestController
+@RequestMapping("/api/kratos")
+class KratosEndpoint
+@Autowired
+constructor(
+        @Autowired private val subjectService: SubjectService,
+        @Autowired private val subjectRepository: SubjectRepository,
+        @Autowired private val projectService: ProjectService,
+        @Autowired private val managementPortalProperties: ManagementPortalProperties,
+) {
+    private var sessionService: SessionService =
+            SessionService(managementPortalProperties.identityServer.publicUrl())
+
+    /**
+     * POST /subjects : Create a new subject.
+     *
+     * @param subjectDto the subjectDto to create
+     * @return the ResponseEntity with status 201 (Created) and with body the new subjectDto, or
+     * with status 400 (Bad Request) if the subject has already an ID
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PostMapping("/subjects")
+    @Timed
+    @Throws(URISyntaxException::class, NotAuthorizedException::class)
+    suspend fun createSubject(
+            @RequestBody webhookDTO: KratosSubjectWebhookDTO,
+    ): ResponseEntity<Void> {
+        logger.debug("REST request to create subject : $webhookDTO")
+        val kratosIdentity =
+                webhookDTO.identity ?: throw IllegalArgumentException("Identity is required")
+
+        if (!kratosIdentity.schema_id.equals(KRATOS_SUBJECT_SCHEMA))
+                throw IllegalArgumentException("Cannot create non-subject users")
+
+        val id = kratosIdentity.id ?: throw IllegalArgumentException("Identity ID is required")
+        val project =
+                kratosIdentity.traits!!.projects?.firstOrNull()
+                        ?: throw NotAuthorizedException("Cannot create subject without project")
+        val projectUserId = project.userId ?: throw IllegalArgumentException("Project user ID is required")
+        val projectDto =
+                projectService.findOneByName(project.id!!)
+                        ?: throw NotFoundException(
+                                "Project not found: ${project.id!!}",
+                                EntityName.PROJECT,
+                                "projectNotFound"
+                        )
+        val subjectDto =
```

Comment:
if possible, try to get the ID from the admin endpoint on kratos and verify with the one in the post request